src/status_im/contexts/onboarding/events.cljs

```diff
@@ -184,27 +183,35 @@
  (fn [{:keys [db]} [auth-method]]
    {:db (assoc db :auth-method auth-method)}))
 
-(rf/defn onboarding-new-account-finalize-setup
-  {:events [:onboarding/finalize-setup]}
-  [{:keys [db]}]
-  (let [masked-password    (get-in db [:onboarding/profile :password])
-        key-uid            (get-in db [:profile/profile :key-uid])
-        syncing?           (get-in db [:onboarding/profile :syncing?])
-        auth-method        (get-in db [:onboarding/profile :auth-method])
-        biometric-enabled? (= auth-method constants/auth-method-biometric)]
-    (cond-> {:db (assoc db :onboarding/generated-keys? true)}
-      biometric-enabled?
-      (assoc :keychain/save-password-and-auth-method
-             {:key-uid         key-uid
-              :masked-password (if syncing?
-                                 masked-password
-                                 (security/hash-masked-password masked-password))
-              :on-success      (fn []
-                                 (rf/dispatch [:onboarding/set-auth-method auth-method])
-                                 (when syncing?
-                                   (rf/dispatch
-                                    [:onboarding/navigate-to-enable-notifications-from-syncing])))
-              :on-error        #(log/error "failed to save biometrics"
-                                           {:key-uid key-uid
-                                            :error   %})}))))
+(rf/reg-event-fx
+ :onboarding/set-reset-display-name
+ (fn [{:keys [db]} [reset-display-name?]]
+   (let [initial-profile-data {:reset-display-name? reset-display-name?
+                               :display-name        (if reset-display-name? "temp username" "")}]
```

Comment:
Added "temp username" because the string cannot be empty while creating a profile, the profile name is updated later when the profile is created.

We can't set it as specified by the issue before, since we don't have the compressed key yet.

Reply:
For such cases it's worth extracting the value to a constant and adding a docstring with a similar explanation as your comment. This way this knowledge won't be lost in GH.